python/src/aiconfig/editor/client/src/components/prompt/model_settings/ModelSettingsRenderer.tsx

```diff
@@ -30,37 +60,45 @@ export default memo(function ModelSettingsRenderer({
   const { classes } = useStyles();
   const [isRawJSON, setIsRawJSON] = useState(schema == null);
 
+  const rawJSONToggleButton = (
+    <Flex justify="flex-end">
+      <JSONEditorToggleButton
+        isRawJSON={isRawJSON}
+        setIsRawJSON={setIsRawJSON}
+      />
+    </Flex>
+  );
+
   return (
     <Flex direction="column" className={classes.settingsContainer}>
-      {/* // TODO: Refactor this out to a generic wrapper for toggling JSONRenderer or children component */}
-      {/* // Only show the toggle if there is a schema to toggle between JSON and custom schema renderer */}
-      {schema && (
-        <Flex justify="flex-end">
-          <Tooltip label="Toggle JSON editor" withArrow>
-            <ActionIcon onClick={() => setIsRawJSON((curr) => !curr)}>
-              {isRawJSON ? (
-                <IconBracesOff size="1rem" />
-              ) : (
-                <IconBraces size="1rem" />
-              )}
-            </ActionIcon>
-          </Tooltip>
-        </Flex>
-      )}
       {isRawJSON || !schema ? (
```

Comment:
Nit: this is a bit different in format from the prompt input (we do something like `isRawJson ? showJson : <ErrorBoundary>...</ErrorBoundary>`. Doesn't have to be this diff but if possible, let's make a note to unify the "format" so the codebase becomes eaiser to read

Reply:
They are a bit different. The input currently renders `PromptInputConfigRenderer` when no schema exists for the prompt model. It's not fully implemented but when it is, we should have the config renderer do some rendering based on the following logic:
- if input exists, implicitly try to match it to the nice renderers that are associated with schema types
- otherwise, render as JSON editor

For settings, we should just default to JSON editor if no schema exists because settings are basically a generic object of any possible key/value pairs